tests/functional/test_model.py

```diff
@@ -0,0 +1,133 @@
+"""Contains tests for the DatabaseModel backends"""
+
+import logging
+import os
+from typing import List, Optional
+
+from databases import Database
+from pydantic import Field
+import pytest
+from sqlalchemy import desc
+
+from pynocular.backends.context import backend
+from pynocular.backends.memory import MemoryDatabaseModelBackend
+from pynocular.backends.sql import SQLDatabaseModelBackend
+from pynocular.model import DatabaseModel
+
+
+@pytest.fixture()
+async def postgres_backend():
+    """Fixture that yields a Postgres backend
+
+    Yields:
+        postgres backend
+
+    """
+    db_host = os.environ.get("DB_HOST", "localhost")
+    db_user_name = os.environ.get("DB_USER_NAME", os.environ.get("USER", "postgres"))
+    db_user_password = os.environ.get("DB_USER_PASSWORD", "")
+    test_db_name = os.environ.get("TEST_DB_NAME", "test_db")
+
+    maintenance_connection_string = f"postgres://{db_user_name}:{db_user_password}@{db_host}:5432/postgres?sslmode=disable"
+    db_connection_string = f"postgresql://{db_user_name}:{db_user_password}@{db_host}:5432/{test_db_name}?sslmode=disable"
+
+    async with Database(maintenance_connection_string) as db:
+        try:
+            await db.execute(f"DROP DATABASE IF EXISTS {test_db_name}")
+            await db.execute(f"CREATE DATABASE {test_db_name}")
+        except Exception as e:
+            logging.info(str(e))
+
+    async with Database(db_connection_string) as db:
+        await db.execute(
+            "CREATE TABLE IF NOT EXISTS things (id SERIAL PRIMARY KEY, name TEXT)"
+        )
+        try:
+            yield SQLDatabaseModelBackend(db)
+        finally:
+            await db.execute("DROP TABLE things")
```

Comment:
Are all the other tests using the SQL backend and if so where is that set at?

Reply:
This is a pytest fixture that sets up a SQL backend with a Postgres database connection. I didn't write any tests for other SQL backends like sqlite. There is a memory_backend fixture defined below that one test uses.

Reply:
I was asking more about how this is set for the other test classes like `test_database_model.py`. I. just assume those will continue to fail until updated? Or was your plan for this module to replace those tests?

Reply:
Gotcha. I didn't touch test_database_model.py yet -- that still uses the original interface and the tests are passing. Now that we're aligned on the high level interface I'll have to go and update tests to use the new backend system or just remove them if test_model.py can be updated to cover the same use cases.